AdobeMediaGallery/Test/Unit/Model/Asset/Command/DeleteByIdTest.php

```diff
@@ -0,0 +1,177 @@
+        $this->adapter = $this->createMock(AdapterInterface::class);
+    }
+
+    /**
+     * Test delete media asset by id action
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testSuccessfulDeleteByIdExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $this->adapter->expects($this->at(0))
+            ->method('delete')
+            ->with($tableName, 'id = ' . $mediaAssetId);
+
+        $this->deleteMediaAssetById->execute($mediaAssetId);
+    }
+
+    /**
+     * Assume that delete action will thrown an Exception
+     *
+     * @param int $mediaAssetId
+     * @dataProvider deleteMediaAssetTestScopes
+     */
+    public function testExceptionOnDeleteExecution(int $mediaAssetId)
+    {
+        $this->getMediaAssetById->expects($this->at(0))
+            ->method('execute')
+            ->with($mediaAssetId)
+            ->willReturn($this->mediaAsset);
+
+        $this->resourceConnection->expects($this->at(0))
+            ->method('getConnection')
+            ->willReturn($this->adapter);
+
+        $tableName = 'media_gallery_asset';
+        $this->resourceConnection->expects($this->at(1))
+            ->method('getTableName')
+            ->with($tableName)
+            ->willReturn($tableName);
+
+        $this->mediaAsset->expects($this->once())
+            ->method('getId')
+            ->willReturn($mediaAssetId);
+
+        $testErrorText = 'test error message';
+        $this->adapter->expects($this->at(0))
+            ->method('delete')
+            ->willThrowException(new \Exception($testErrorText));
+        $this->expectException(CouldNotDeleteException::class);
+        $testMessage = __(
+            'Could not delete media asset with id %id: %error',
+            [
+                'id' => $mediaAssetId,
+                'error' => $testErrorText,
+            ]
+        )->render();
+        $this->logger->expects($this->once())
+            ->method('critical')
+            ->with($testMessage)
+            ->willReturnSelf();
+        $this->expectExceptionMessage($testMessage);
+        $this->deleteMediaAssetById->execute($mediaAssetId);
+    }
+
+    /**
+     * @return array
+     */
+    public function deleteMediaAssetTestScopes(): array
```

Comment:
Data provider is not necessary to providing id's in my opinion, the behaviour is pretty much the same for any `id`.

Also, (note for the future unit tests) data providers are usually used to provide at least two arguments for the test:
test data and expected result (or flags telling the test how the verification should change depending on test data provided) i.e. `$input, $output` or `$input, $isExceptionExpected`